plugins/catalog-backend/migrations/20231009175335_auto_incrementing_pk.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function up(knex) {
+  await knex.schema.hasColumn('refresh_keys', 'id').then(async hasCol => {
+    if (hasCol) {
+      return;
+    }
+    await knex.schema.alterTable('refresh_keys', table => {
+      table
+        .uuid('id')
```

Comment:
What is distinguishing between the cases where `id` is a `uuid` vs `.increments('id', { primaryKey: true })` (presumably an integer?).

Reply:
I found this commit: https://github.com/backstage/backstage/pull/20409/commits/c5ef5e36df8ed1c7e1e884aecfbe99df236d6529

Can you give more context for this? The database should be able to enforce monotonically increasing primary keys. That's pretty standard database behavior.

Reply:
We've found that with knex uuid non-integer columns don't auto-increment. It looks like perhaps we can use the uuid function as part of the column definition. @xandroc can you look into that? https://stackoverflow.com/questions/61349754/how-to-generate-autoincrement-guid-on-insert-without-triggers-and-manual-inserts

Reply:
I'm mostly just curious why some tables have `uuid` primary keys and other have `increments` primary keys. If the answer is just "Backstage plugins are inconsistent", then that's good enough for me.